tests/rpc-tests/rpc-compat.js

```diff
@@ -0,0 +1,199 @@
+  const provider = new HDWalletProvider({
+    mnemonic: {
+      phrase: process.env.MNEMONIC
+    },
+    providerOrUrl: `http://${machine}:8545`
+  })
+
+    return new Web3(provider)
+}
+
+async function init() {
+  console.log('📍Executing RPC compatibility tests')
+  const doc = await loadDevnetConfig('remote')
+  let machine = doc.devnetBorHosts[0]
+  let user = doc.devnetBorUsers[0]
+  let ip = `${user}@${machine}`
+  const borStartScriptLocation = '~/node/bor-start.sh'
+
+  const walletDisableFlagCmd = `grep -q 'disable-bor-wallet=false' ${borStartScriptLocation} && echo 'found' || echo 'not found'`
+  const isPresent = await runSshCommandWithReturn(ip, walletDisableFlagCmd, maxRetries)
+ 
+  if (isPresent === 'not found') {
+    const addFlagsCmd = `sed -i 's/--allow-insecure-unlock \\\\/&\\n  --disable-bor-wallet=false \\\\/' ${borStartScriptLocation}`
+    const restartBorCmd = 'sudo service bor restart'
+    const isSyncingCmd = `~/go/bin/bor attach ~/.bor/data/bor.ipc --exec "eth.syncing"`
+  
+    console.log('📍Updating start script on machine to unlock node account ... ')
+    await runSshCommand(ip, addFlagsCmd, maxRetries)
+    
+    console.log('📍Restarting bor on machine ... ')
+    await runSshCommand(ip, restartBorCmd, maxRetries)
+  
+    await timer(100000)
+  
+    while (true) {
+      var isSyncing = await runSshCommandWithReturn(ip, isSyncingCmd, maxRetries)
+      if (isSyncing === 'false') {
+        break
+      }
+    }
+  }
+
+  const web3 = await initWeb3(machine)
+
+  let src = `${ip}:~/.bor/address.txt`
+  let dest = './address.txt'
+  await runScpCommand(src, dest, maxRetries)
+
+  const addrText = fs.readFileSync('./address.txt', 'utf-8')
+  const nodeAccount = addrText.split(/\r?\n/)[0]
+  const balance = await web3.eth.getBalance(nodeAccount)
+
+  if (balance <= web3.utils.toWei('0.5', 'ether')) {
+    console.log("Node account is running low on funds. Refilling...")
+    await web3.eth.accounts.wallet.add(process.env.PRIVATE_KEY)
+    const sender = web3.eth.accounts.privateKeyToAccount(process.env.PRIVATE_KEY)
+    const nonce = await web3.eth.getTransactionCount(sender.address)
+    await fundAccount(web3, sender, [nodeAccount], 30000000009, 30000000000, nonce)
+  }
+  console.log("Balance of node account: ", balance)
+
+  return [machine, web3, nodeAccount]
+  
+}
+
+function fetchTestFiles(dir) {
+  try {
+    let testData = []
+
+    fs.readdirSync(dir).forEach((file) => {
+      const filePath = path.join(dir, file)
+      const stat = fs.statSync(filePath)
+
+      if (stat.isDirectory()) {
+        testData = testData.concat(fetchTestFiles(filePath))
+      } else {
+        const data = fs.readFileSync(filePath, 'utf8')
+        const jsonData = JSON.parse(data)
+        testData.push(jsonData)
+      }
+    })
+
+    return testData
+  } catch (error) {
+    console.error('Error occured while fetching test rpc data: ', error)
+  }
+}
+
+export async function rpcCompat() {
```

Comment:
What do you mean with Compat? :)

Reply:
Yeah, poor wording from me 😓

Reply:
`rpcTest` should be good enough I guess. I most probably had started writing this with compatibility in mind but that's not the only thing we're checking for.